stdlib/public/core/OutputStream.swift

```diff
@@ -330,11 +330,14 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
-    for c in string.utf8 {
-      _swift_stdlib_putchar_unlocked(Int32(c))
+    let utf8 = string.nulTerminatedUTF8
+    utf8.withUnsafeBufferPointer {
+        (utf8) -> Void in
+        let result = _swift_stdlib_fwrite_stdout(UnsafePointer(utf8.baseAddress),
+            utf8.count - 1, 1)
+        if result < 0 {
```

Comment:
Thanks.  Looks like `if result != 1` is not good enough because sometimes `write()` is called for the empty string.  In that case, `utf8.count` is `1` so a `size` of `0` is passed to `fwrite()`, causing `0` to be returned.  I'll rework it, there's no point in calling `fwrite()` if there's nothing to write.